frontend/src/main/java/starter/project/frontend/CarController.java

```diff
@@ -42,20 +27,23 @@ public Car getCar(@PathVariable("carId") int carId) {
 
     @PostMapping(path = "/cars", consumes = {"application/json"})
     public Map<Integer, Car> createCarList(@RequestBody List<Car> carList) {
-        List<Car> currentCarList = new ArrayList<>();
-        currentCarList = carList;
+
         for (int i = 0; i < carList.size(); i++) {
-            map.put(i, currentCarList.get(i));
-            currentCarList.get(i).setId(i);
+            map.put(carList.get(i).getId(), carList.get(i));
         }
         return map;
     }
 
+    @PostMapping(path = "/car", consumes = {"application/json"})
+    public Map<Integer,Car> addCar(@RequestBody Car car) {
+        map.put(car.getId(),car);
+        return map;
+    }
+
+
     @PutMapping(path = "car", consumes = {"application/json"})
     public Map<Integer, Car> updateCars(@RequestBody Car car) {
-        Car entry = map.get(car.getId());
-        entry = car;
-        map.put(entry.getId(), entry);
+        map.put(car.getId(), car);
```

Comment:
One last change, in `batchUpdateCars` we also have the same problem as before with the unused variable. A similar change like you did here would be good.